packages/stack/blockchain/src/index.js

```diff
@@ -22,57 +21,43 @@ class Blockchain {
     embark.registerActionForEvent("pipeline:generateAll:before", this.addArtifactFile.bind(this));
 
     this.blockchainNodes = {};
-    this.events.setCommandHandler("blockchain:node:register", (clientName, startCb) => {
-      this.blockchainNodes[clientName] = startCb;
+    this.events.setCommandHandler("blockchain:node:register", (clientName, { isStartedFn, launchFn, stopFn }) => {
+      this.blockchainNodes[clientName] = { isStartedFn, launchFn, stopFn };
     });
 
-    this.events.setCommandHandler("blockchain:node:start", async (initialBlockchainConfig, cb) => {
-      this.plugins.emitAndRunActionsForEvent("blockchain:config:modify", initialBlockchainConfig, (err, blockchainConfig) => {
+    this.events.setCommandHandler("blockchain:node:start", (blockchainConfig, cb) => {
+      const self = this;
```

Comment:
Looks like this code has mostly been moved around, but maybe we can use the opportunity to get rid off `self = this`.

Reply:
Removed.